spec/puppet/resource_api/value_creator_spec.rb

```diff
@@ -0,0 +1,224 @@
+require 'spec_helper'
+
+RSpec.describe Puppet::ResourceApi::ValueCreator do
+  subject(:value_creator) do
+    described_class
+  end
+
+  let(:resource_class) { Puppet::ResourceApi::Property }
+  let(:data_type) { Puppet::Pops::Types::PEnumType.new(['absent', 'present']) } # rubocop:disable Style/WordArray
+  let(:param_or_property) { :newproperty }
+  let(:options) do
+    {
+      type: 'Enum[present, absent]',
+      desc: 'Whether this resource should be present or absent on the target system.',
+      default: 'present',
+    }
+  end
+
+  before(:each) do
+    allow(resource_class).to receive(:newvalue)
+    allow(resource_class).to receive(:newvalues)
+    allow(resource_class).to receive(:aliasvalue)
+    allow(resource_class).to receive(:defaultto)
+    allow(resource_class).to receive(:isnamevar)
+  end
+
+  it { expect { described_class.create_values(nil) }.to raise_error ArgumentError, %r{wrong number of arguments} }
+  it { expect { value_creator }.not_to raise_error }
+
+  describe '#create_values' do
+    before(:each) do
+      value_creator.create_values(resource_class, data_type, param_or_property, options)
+    end
+
+    context 'when resource class is property' do
+      it 'resource_class has #call_provider defined' do
+        expect(resource_class.method_defined?(:call_provider)).to eq(true)
+      end
+
+      context 'when property is :ensure' do
+        it 'calls #newvalue twice' do
+          expect(Puppet::ResourceApi::Property).to have_received(:newvalue).twice
```

Comment:
All these tests are very shallow. At least also test the received arguments using `with()` matchers (see https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/setting-constraints/matching-arguments).

Reply:
I've updated specs file :slightly_smiling_face: